In [5]:
'''
Doing dot product with a layer of neurons and multiple inputs
Associated YT NNFS tutorial: https://www.youtube.com/watch?v=tMrbN67U9d4
'''

import numpy as np 

inputs = [1.0, 2.0, 3.0, 2.5]
weights = [[0.2, 0.8, -0.5, 1.0],
           [0.5, -0.91, 0.26, -0.5],
           [-0.26, -0.27, 0.17, 0.87]] #layer of 3 neurons, matrix of vectors

biases = [2.0, 3.0, 0.5]

output = np.dot(weights, inputs) + biases
print(output)

[4.8   1.21  2.385]


The calculations being done to produce the outputs 

inputs = [1.0, 2.0, 3.0, 2.5] 
weights = [0.2, 0.8, -0.5, 1.0]
bias = 2.0

output = np.dot(weights, inputs) + biases

a.b = [1.0, 2.0, 3.0, 2.5].[0.2, 0.8, -0.5, 1.0] = 0.2*1.0 + 0.8*2.0 + -0.5*3.0 + 1.0*2.5 = 2.8 then 2.8 + 2.0(0th/first bias) = 4.8

The dot product of two vectors gives a scalar single value
The dot product calculation is then done for the matrix of vectors
weights is a matrix of three vectors it will return an array of values that will be the dot product of the first weights and inputs the second weights and inputs and the third weights and inputs then add these vector values with the vector values of biases to get the three final output values of [4.8, 1.21, 2.385]
​
next is part 4

output = weight.input + bias
y = mx + c
imagine a streaight line in a 2D graph y axis and x axis 
Activation functions or activation points ReLU
output = ReLU(weight.input + bias)